File: ClientLeger/app/Models/CompoCommande.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class CompoCommande extends Model
{
    use HasFactory;

    protected $table = 'compo_commandes';
    protected $primaryKey = ['id_commande_ligne', 'id_ingredient'];
    public $incrementing = false;
    public $timestamps = false;

    protected $fillable = [
        'id_commande_ligne',
        'id_ingredient',
        'prix'
    ];

    /**
     * Get the order line that this ingredient belongs to.
     */
    public function commandeLigne()
    {
        return $this->belongsTo(Commande_ligne::class, 'id_commande_ligne', 'id_commande_ligne');
    }

    /**
     * Get the ingredient information.
     */
    public function ingredient()
    {
        return $this->belongsTo(Ingredient::class, 'id_ingredient', 'id_ingredient');
    }
} 

File: ClientLourd/TestFX/src/application/model/CompoCommande.java
package application.model;

public class CompoCommande {
    private int id_commande_ligne;
    private int id_ingredient;
    private double prix;

    public CompoCommande(int id_commande_ligne, int id_ingredient, double prix) {
        this.id_commande_ligne = id_commande_ligne;
        this.id_ingredient = id_ingredient;
        this.prix = prix;
    }

    public int getId_commande_ligne() { return id_commande_ligne; }
    public int getId_ingredient() { return id_ingredient; }
    public double getPrix() { return prix; }
} 